monarchmoney/monarchmoney.py

```diff
@@ -2352,14 +2392,54 @@ async def set_budget_amount(
             graphql_query=query,
         )
 
+    async def upload_and_parse_balance_history(
+        self,
+        account_id: str,
+        csv_content,
+        timeout: int = 300,
+        delay: int = 10,
+    ) -> bool:
+        """
+        Uploads and parses the balance history, updating the account balance on monarch money
+
+        :param account_id: The account ID to apply the history to.
+        :param csv_content: CSV representation of the balance history. Headers are Date, Amount, and Account Name.
+        :param timeout: The number of seconds to wait before timing out
+        :param delay: The number of seconds to wait for each check on whether parsing is completed
+        """
+
+        session_key = await self.upload_account_balance_history(
+            account_id=account_id, csv_content=csv_content
+        )
+
+        response = await self.parse_upload_balance_history_session(
+            session_key=session_key
+        )
+
+        is_completed = (
+            response["parseBalanceHistory"]["uploadBalanceHistorySession"]["status"]
+            == "completed"
+        )
+
+        start = time.time()
+
+        while not is_completed and (time.time() <= (start + timeout)):
+            await asyncio.sleep(delay)
+
+            is_completed = (await self.get_upload_balance_history_session(session_key))[
+                "uploadBalanceHistorySession"
+            ]["status"]
```

Comment:
I believe this is missing a `== "completed"` check. 

Though, I believe a better approach would be to check whether the processing is still "pending" so that this code short circuits if the upload fails.

Reply:
Yes, it is missing the check! Thanks for catching that!

The possible statuses that I've found so far are: "created", "started", and "completed". After calling `_initiate_upload_balance_history_session`, it usually returns the "created" status. Then `_is_upload_balance_history_complete` can either return the "started" or "completed" statuses.

If `_is_upload_balance_history_complete` can return "created", it may lead to a premature completed check. Though I don't think that's likely. 

@hammem , thoughts on whether we should check for "started" vs "completed"? I could see the better case being checking for "started" but worried about possible side effects that may cause.

Reply:
Ah I was expecting a failed state if an invalid csv is provided. In that case, looks like the following error is thrown. Which I think is sufficient. Please disregard the suggestion!

```
gql.transport.exceptions.TransportQueryError: {'message': "Something went wrong while processing: ['parseBalanceHistory']
```